services/wallet/service.go

```diff
@@ -34,7 +34,7 @@ type Service struct {
 	db                  *Database
 	reactor             *Reactor
 	signals             *SignalsTransmitter
-	client              *ethclient.Client
+	client              reactorClient
```

Comment:
I notice that the previous implementation used a pointer to an `*ethclient.Client`, did you deliberately chose to not have a struct pointer for your reactorClient? This comment also relates to the other cases in this PR.

Reply:
`reactorClient` was an interface, not a struct and thus pointer wouldn't make sense there as I understand, right? But it was useless there and I replaced it in the last commit with `*walletClient` which is a struct pointer.